### PR TITLE
feat: add UIN validation option

### PR DESCRIPTION
**Pull request by dalsontws**
[Notion link](notion-link-here) 

This PR adds... 

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [x] I've requested reviews from 1-2 reviewers
~I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components~
~I've added/updated unit/integration tests~
~I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow~
~I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)~
~I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)~

### CODE REVIEW

File: src/utils/validateUin.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    return false;
+  }
+
+  return true;
+};
+
+/**
+ * This function is called to clean the input UIN and
+ * validate the specific UIN using the regex check
+ * and our checksum.
```
Comment (ilhaoni) [inline src/utils/validateUin.tsx:31]: can be more generic here 😂 
```suggestion
 * and checksum.
```

**Review by ilhaoni — COMMENTED** [review]
Reviewed! just some small nits xD

File: src/types.ts [hunk 2]
```diff
@@ -136,6 +136,7 @@ const IdentificationFlag = t.intersection([
       t.literal("NRIC"),
       t.literal("PASSPORT"),
       t.literal("REGEX"),
+      t.literal("UIN_REGEX"),
```
[the remark this answers is not in the document]
Comment (dalsontws) [reply src/types.ts:139]: oh actually i was thinking cos its UIN + REGEX so `UIN_REGEX`, but i think `UIN` works too! lemme change this!

File: src/utils/validateUin.tsx (hunk 1 above)
Comment (dalsontws) [reply src/utils/validateUin.tsx]: changed!

**Review by ilhaoni — APPROVED** [review]
LGTM! ^^

**Pull request merged**